test/e2e/framework/util.go

```diff
@@ -1284,3 +1285,110 @@ func taintExists(taints []v1.Taint, taintToFind *v1.Taint) bool {
 	}
 	return false
 }
+
+// WatchEventSequenceVerifier ...
+// manages a watch for a given resource, ensures that events take place in a given order, retries the test on failure
+//   testContext         cancelation signal across API boundries, e.g: context.TODO()
+//   dc                  sets up a client to the API
+//   resourceType        specify the type of resource
+//   namespace           select a namespace
+//   resourceName        the name of the given resource
+//   listOptions         options used to find the resource, recommended to use listOptions.labelSelector
+//   expectedWatchEvents array of events which are expected to occur
+//   scenario            the function to run
+func WatchEventSequenceVerifier(testContext context.Context, dc dynamic.Interface, resourceType schema.GroupVersionResource, namespace string, resourceName string, listOptions metav1.ListOptions, expectedWatchEvents []watch.Event, scenario func(*watchtools.RetryWatcher) []watch.Event) {
+	initResource, err := dc.Resource(resourceType).Namespace(namespace).List(testContext, listOptions)
+	ExpectNoError(err, "Failed to fetch initial resource")
+	listWatcher := &cache.ListWatch{
+		WatchFunc: func(listOptions metav1.ListOptions) (watch.Interface, error) {
+			return dc.Resource(resourceType).Namespace(namespace).Watch(testContext, listOptions)
+		},
+	}
+	resourceWatch, err := watchtools.NewRetryWatcher(initResource.GetResourceVersion(), listWatcher)
+	ExpectNoError(err, "Failed to create a resource watch of %v in namespace %v", resourceType.Resource, namespace)
+
+	// NOTE value of 3 retries seems to make sense
+	retries := 3
```

Comment:
I would make this configurable with a sane default

Reply:
a comment to explain why we're retrying up top vs. having to spelunk into the loop would be appreciated

Reply:
I can make it as an input, but I'm unsure how to make it a default. Would there be any insights on that?

Reply:
I've updated the docs for the function in 5268ba5488c.

I'd be happy to submit a later PR to make the retry count configurable